shared/src/main/scala/mlscript/Codegen.scala

```diff
@@ -200,9 +205,15 @@ final case class JSPlaceholderExpr() extends JSExpr {
 final case class JSArrowFn(params: Ls[JSPattern], body: JSExpr) extends JSExpr {
   def precedence: Int = 22
   def toSourceCode: SourceCode =
-    ((params map { _.toSourceCode }) reduceLeft [SourceCode] { case (x, y) =>
-      x ++ SourceCode.from(", ") ++ y
-    }).parenthesized ++ SourceCode.from(" => ") ++ body.toSourceCode
+    params
+      .foldLeft(SourceCode.empty) { case (x, y) =>
+        x ++ SourceCode.from(", ") ++ (y match {
+          // JavaScript does not allow empty arguments, e.g. `function (,x) {}`.
```

Comment:
I don't understand what that comment is trying to say.
Won't your function wrongly generate `(,x) => ...` for 1-parameter lambdas?
Instead, it should generate either `(x) => x + 1` or `(x,) => x + 1` (both are valid).

Reply:
That comment is for the line below `case JSWildcardPattern() => SourceCode.from("_")`. Because we translate literals to wildcards, and JavaScript does not allow holes in parameter list, e.g. `function f(,,y)`.

But I made a mistake during refactoring. `SourceCode.from(", ")` should be after `y match { ... }`. Thanks for pointing out. And unfortunately I didn't test functions. (I just tested `def x = 1 + 1`)

Reply:
Okay. The generated code will not be valid though if there is more than one wildcard or literal: JS rejects `(x,_,_) => x + 1`. We may need to zip the parameters with their index (`.zipWithIndex`) and generate `(x,_1,_2) => x + 1` instead. And do something similar for `let` patterns.

Reply:
Good suggestion! Just updated!